test/conftest.py

```diff
@@ -103,8 +105,30 @@ def backend(request, backend_name, xephyr, wayland_session):
 
 
 @pytest.fixture(scope="function")
-def manager_nospawn(request, backend):
+def log_queue():
+    """Creates a new Queue for logging messages run in a backend process."""
+    yield multiprocessing.Queue(-1)
+
+
+@pytest.fixture(scope="function")
+def logger(caplog, log_queue):
+    """
+    Connects logging messages in the backend to a logger in the main thread
+    and returns a caplog fixture  which can access those messages.
+    """
+    root = logging.getLogger()
+    listener = logging.handlers.QueueListener(log_queue, *root.handlers)
+    listener.start()
+
+    yield caplog
+
+    listener.stop()
+
+
+@pytest.fixture(scope="function")
+def manager_nospawn(request, backend, log_queue):
     with TestManager(backend, request.config.getoption("--debuglog")) as manager:
+        manager.log_queue = log_queue
```

Comment:
Is there a reason it's only enabled for `manager_nospawn` and not `manager` too?

Reply:
Because `manager` is just `manage_nospawn` that is started before the fixture is yielded.

Reply:
doh! :laughing: In that case, `TestManager` instances always have the log_queue attribute, making the `hasattr` check elsewhere redundant, right?

Reply:
Not quite. The `xmanager` fixture also creates a `TestManager` instance but I haven't added the log queue there as that seemed to be a different use case. Without the `hasattr` check, any tests using that fixture would fail.

Reply:
I see - OK no problem